File: Sorting/SelectionSort/selectionSort.cpp
#include<iostream>
#include <time.h>

using namespace std;


int main(){

	clock_t t1,t2;
    t1 = clock();

	int n;
	cout<<"Enter the size you want the array to be: ";
	cin>>n;
	int arr[n];
	cout<<"Enter " <<n<< " numbers for the array";

	for(int i = 0; i < n; i++){
		cin>>arr[i];
	}

	for(int i = 0; i < n-1; i++){
		int min = i;

		for(int j = i+1; j < n; j++){
			if(arr[min] > arr[j]){
				min = j;
			}
		}

		if(min != i){	
			int temp = arr[i];
			arr[i] = arr[min];
			arr[min] = temp;
		}
	}

	cout<<"Sorted array is: ";
	for(int i = 0; i < n; i++){
		cout<<arr[i]<<"\t";
	}

	t2 = clock();
    float diff ((float)t2-(float)t1);
    float secs = diff/1000;
    cout<<"\nTIME TAKEN:" <<secs<<endl;

}

File: Sorting/SelectionSort/SelectionSort.java
import java.util.*;
public class SelectionSort{

	public static void main(String[] args){
		Random r = new Random();
		int arr[] = new int[r.nextInt(40)];
		for(int j = 0; j < arr.length; j++){
			arr[j] = r.nextInt(5000);		
		}
		selectionSort(arr);
	}

	public static void selectionSort(int arr[]){
		for(int i = 0; i < arr.length - 1; i++){
			int min = i;
			for(int j = i + 1; j < arr.length; j++){
				if(arr[j] < arr[min]){
					min = j;
				}
			}
			if(i != min){
				int temp = arr[i];
				arr[i] = arr[min];
				arr[min] = temp;
			}
		}
		
		for(int i = 0; i < arr.length; i++){
			System.out.print(arr[i] + ", ");
		}
	}

}
